ethereum/evmtool/src/main/java/org/hyperledger/besu/evmtool/T8nExecutor.java

```diff
@@ -270,6 +270,7 @@ static T8nResult runTest(
       final TransactionProcessingResult result;
       try {
         tracer = tracerManager.getManagedTracer(i, transaction.getHash());
+        tracer.traceStartTransaction(worldStateUpdater, transaction);
```

Comment:
I am not sure why the tracer was not called here previously?

Reply:
it seems you added the traceEndTransaction @diega . do you know if it's normal to not have the startTransaction ?

Reply:
Yes, it looks like it was needed. I never used the `--json|--trace` flag from the `t8n` tool, so I cannot say if/how it was working without the tracer start. Probably @shemnon can confirm it

Reply:
It feels like this should be called somewhere in [MainnetTransactionProcessor.processTransaction] (https://github.com/hyperledger/besu/blob/main/ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/MainnetTransactionProcessor.java#L260) rather than as part of the lead-up.

Reply:
same with endTransaction.

Reply:
@shemnon given that `gasUsed` can't be computed from within `processTransaction`, that implies to either:

 - dirty `processTransaction` signature to pass `gasUsed` only for the sake of tracing;
 - just drop `gasUsed` from the `traceEndTransaction` API.

What do you think?